sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/utils/MqGrayscaleConfigUtils.java

```diff
@@ -0,0 +1,294 @@
+        } else {
+            boolean isAllowRefresh = isAllowRefreshChangeFlag(CACHE_CONFIG.get(CACHE_CONFIG_KEY), config);
+            if (isAllowRefresh) {
+                MqGrayscaleConfig cacheConfig = CACHE_CONFIG.get(CACHE_CONFIG_KEY);
+                cacheConfig.updateGrayscaleConfig(config);
+                SubscriptionDataUtils.updateChangeFlag();
+            }
+        }
+    }
+
+    private static void buildGrayTagsSet(MqGrayscaleConfig config) {
+        for (GrayTagItem item : config.getGrayscale()) {
+            if (!item.getTrafficTag().isEmpty()) {
+                GRAY_TAGS_SET.addAll(item.getTrafficTag().keySet());
+            }
+        }
+    }
+
+    /**
+     * only traffic label changes allow refresh tag change map to rebuild SQL92 query statement,
+     * because if the serviceMeta changed, the gray consumer cannot be matched and becomes a base consumer
+     * so, if you need to change the env tag, restart all services.
+     *
+     * @param resource cache config
+     * @param target cache config
+     * @return boolean
+     */
+    private static boolean isAllowRefreshChangeFlag(MqGrayscaleConfig resource, MqGrayscaleConfig target) {
+        if (resource.isBaseExcludeGroupTagsChanged(target)) {
+            return true;
+        }
+        if (resource.isConsumerTypeChanged(target)) {
+            return true;
+        }
+        return !resource.buildAllTrafficTagInfoToStr().equals(target.buildAllTrafficTagInfoToStr());
+    }
+
+    /**
+     * get plugin enabled
+     *
+     * @return plugin enabled
+     */
+    public static boolean isPlugEnabled() {
+        return CACHE_CONFIG.get(CACHE_CONFIG_KEY).isEnabled();
+    }
+
+    /**
+     * compare serviceMeta with mqGrayscaleConfig, set message property
+     *
+     * @param message message
+     */
+    public static void setUserPropertyByServiceMeta(Message message) {
+        if (!CACHE_CONFIG.get(CACHE_CONFIG_KEY).isEnabled()) {
+            return;
+        }
+        MqGrayscaleConfig mqGrayscaleConfig = CACHE_CONFIG.get(CACHE_CONFIG_KEY);
+        if (mqGrayscaleConfig == null) {
+            return;
+        }
+        Map<String, String> grayTags = mqGrayscaleConfig.getGrayTagsByServiceMeta(MICRO_SERVICE_PROPERTIES);
+        if (grayTags.isEmpty()) {
+            return;
+        }
+        for (Map.Entry<String, String> entry : grayTags.entrySet()) {
+            message.putUserProperty(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * compare traffic properties with mqGrayscaleConfig, set message property
+     *
+     * @param message message
+     */
+    public static void setUserPropertyByTrafficTag(Message message) {
+        if (!CACHE_CONFIG.get(CACHE_CONFIG_KEY).isEnabled() || getTrafficTag().isEmpty()) {
+            return;
+        }
+        MqGrayscaleConfig mqGrayscaleConfig = CACHE_CONFIG.get(CACHE_CONFIG_KEY);
+        if (mqGrayscaleConfig == null) {
+            return;
+        }
```

Comment:
Check whether the value is null and then check whether the switch is enabled.

Reply:
static{} has set value, will not null